rest_framework/permissions.py

```diff
@@ -172,7 +178,25 @@ def get_required_object_permissions(self, method, model_cls):
         return [perm % kwargs for perm in self.perms_map[method]]
 
     def has_object_permission(self, request, view, obj):
-        model_cls = view.queryset.model
+        getter = getattr(view, 'get_queryset', None)
+        if getter is not None:
+            try:
+                queryset = view.get_queryset()
+            except AssertionError:
+                # The default implementation of view.get_queryset()
+                # didn't find the view.queryset attribute.
+                queryset = None
+        else:
+            try:
+                queryset = view.queryset
+            except AttributeError:
+                queryset = None
```

Comment:
Same for this block, would be simpler as this, no?...

    try:
        queryset = view.get_queryset()
    except AttributeError:
        queryset = getattr(view, 'queryset', None)